instrumentation/spring-4.3.0/src/main/java/com/nr/agent/instrumentation/SpringControllerUtility.java

```diff
@@ -17,6 +17,8 @@
 
 public class SpringControllerUtility {
 
+    public static String placeholderValue;
```

Comment:
This strategy isn't threadsafe and needs some rethinking.

The flow works as follows:

`UrlPathHelper#getLookupPathForRequest` calls `SpringControllerUtility#setPlaceholderValue`, which sets `SpringControllerUtility.placeholderValue` to be path.

Later in the flow, `SpringControllerUtility#getPath` checks if `SpringControllerUtility.placeholderValue` has a value, and if so, uses it to construct the transaction path.

The problem is that multiple threads are sharing the single static instance of `SpringControllerUtility#placeholderValue` to maintain state. I've spun up a test app locally and setup a test to execute requests in parallel and have demonstrated that this produces corrupt transactions.

The logic also breaks if only a single thread is used to make requests. Suppose you have two routes `GET /route1` and `GET /${ENV_VAR}`, and an env var named `ENV_VAR` is set to `route2` so that you effectively have `GET /route2`. If you call `GET /route2`, `SpringControllerUtility#placeholderValue` will be set to `route2`, and subsequent calls to `GET /route1` will see the value of `route2` in `SpringControllerUtility#placeholderValue` and report their transaction path as `/route2`, which is not good.

Reply:
Ah, understand. Thank you for the feedback!